src/components/Comments/Comment.js

```diff
@@ -67,12 +67,19 @@ const Comment = ({
   const {createSignChat} = useCreateChat();
   const {sendMessageDM} = useDMMessage();
 
+  const userName =
+    (comment.data?.anon_user_info_color_name
+      ? getOfficialAnonUsername(comment?.data)
+      : user?.data?.username) +
+    (comment.is_you ? ' (You)' : '') +
+    (comment.is_author ? ' (Post Author)' : '');
+
   const onTextPress = () => {
     if (level >= 2 || disableOnTextPress) {
       return;
     }
     if (onPress && typeof onPress === 'function') {
-      onPress();
+      onPress(comment.id, userName);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for generating `userName` has been updated to include additional conditions and concatenations. Make sure these changes are reflected in the unit tests. Also, ensure that the `onPress` function now correctly handles the new parameters `comment.id` and `userName`.
```diff
+   const userName =
+     (comment.data?.anon_user_info_color_name
+       ? getOfficialAnonUsername(comment?.data)
+       : user?.data?.username) +
+     (comment.is_you ? ' (You)' : '') +
+     (comment.is_author ? ' (Post Author)' : '');
+ 
+    const onTextPress = () => {
+      if (level >= 2 || disableOnTextPress) {
+        return;
+      }
+      if (onPress && typeof onPress === 'function') {
+       onPress(comment.id, userName);
+     }
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->